@shared/api/external.ts

```diff
@@ -31,25 +37,23 @@ export const requestPublicKey = async (): Promise<string> => {
     console.error(e);
   }
 
-  const { publicKey, error } = response;
-
-  if (error) {
-    throw error;
-  }
-  return publicKey;
+  return { publicKey: response?.publicKey || "", error: response?.apiError };
```

Comment:
A similar change happens in each of the functions below:

instead of `throw`ing on error, we're returning the error as a possible key on the payloaf